### PR TITLE
fix: rich select keyboard navigation not working

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

https://app.clickup.com/t/1jv4f5v

Dependant of: https://github.com/ArkEcosystem/nodem/pull/277

refactor the way it gets the options count (used in the keyboard navigation) since the current implement loses the value after livewire re-renders the content.

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [x] I provided a screenshot of my changes to the component _(if applicable)_
-   [x] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: resources/assets/js/rich-select.js [hunk 1]
```diff
@@ -12,16 +12,21 @@ const RichSelect = (
...
+
+        if (grouped) {
+            this.optionsCount = Object.keys(this.options)
+                .map((groupName) => Object.keys(this.options[groupName]))
+                .flat().length;
+        } else {
+            this.optionsCount = Object.keys(this.options).length;
+        }
+
+        console.log(this.options);
```
Comment (alexbarnsley) [inline resources/assets/js/rich-select.js:28]: debug

**Review by alexbarnsley — CHANGES_REQUESTED** [review]
Using the keyboard to go down still seems to go only as far as the second option :thinking: if i then go back up, then try to go up again (e.g. selected index -1) i get a console error:

image

~Added video to clickup card~

Nevermind - I missed the nodem PR!

File: resources/assets/js/rich-select.js [hunk 2]
```diff
@@ -93,13 +97,15 @@ const RichSelect = (
...
         this.selected =
-            this.selected - 1 < 0 ? this.optionsCount - 1 : this.selected - 1;
+            this.selected - 1 < 0 ? optionsCount - 1 : this.selected - 1;
         this.scrollToSelectedOption();
     },
     onArrowDown() {
+        const optionsCount = this.getOptionsCount();
         this.selected =
-            this.selected + 1 > this.optionsCount - 1 ? 1 : this.selected + 1;
+            this.selected + 1 > optionsCount - 1 ? 1 : this.selected + 1;
```
Comment (alexbarnsley) [inline resources/assets/js/rich-select.js:108]: this should reset to 0, not 1

```suggestion
            this.selected + 1 > optionsCount - 1 ? 0 : this.selected + 1;
```

**Review by alexbarnsley — CHANGES_REQUESTED** [review]
As per the comment, when going "up" to the last item in the list, then "down" to the first, it actually goes to the second (index 1). I've posted a video in the clickup card

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**